Add tests for loading JWST data

drdavella opened the pull request:
This resolves #460.

SaOgaz commented:
Hey @drdavella, I've got another cubeviz test file ready to go (already checked cubeviz can open it fine). Do you have time to add it to this PR or should I open a new issue for it?

filename: ditherunity_CLEAR_PRISM_M1_m1_noiseless_NRS1_modified_updatedHDR_fixintarget_s3d.fits

drdavella commented:
@SaOgaz I added a new developer documentation section with instructions for testing. I also added some basic documentation about building the docs while I was at it.

drdavella commented:
I'm pretty sure this is ready for final review and merge. I believe the reason the coverage has decreased for this PR is because I added a commit that removed the test files from coverage (which should have been the case to begin with). This means technically the coverage has gone down, but future PRs should be more accurate.
